LibraBFT/Impl/Consensus/SafetyRules/SafetyRules.agda

```diff
@@ -199,3 +220,28 @@ module constructAndSignVoteM-continue2 (voteProposal : VoteProposal) (validatorS
         ok vote
 
 constructAndSignVoteM-continue2 = constructAndSignVoteM-continue2.step₀
+
+------------------------------------------------------------------------------
+
+signProposalM : BlockData → LBFT (Either ErrLog Block)
+signProposalM blockData = do
+ vs ← use (lSafetyRules ∙ srValidatorSigner)
+ maybeS vs (bail fakeErr) {-ErrL (here' ["srValidatorSigner", "Nothing"])-} $ λ validatorSigner -> do
+  safetyData ← use (lPersistentSafetyStorage ∙ pssSafetyData)
+  verifyAuthorM (blockData ^∙ bdAuthor) ∙?∙ λ unit →
```

Comment:
I know this is a really small issue, but any reason not to use `_` for the bound `unit`s?

Reply:
Yes.   `_` will match anything, whereas `unit` will only match `unit` and that is the only possible thing in this case.   So tighter type-checking.

Reply:
(I see you updated the Haskell code so that it pattern matches on unit, so that these two versions are the same -- which is the most important aspect of the comment).

That's not what's happening here -- you are binding a *variable* named unit. Add a `where` directly after the lambda to get pattern matching.

Reply:
Good catch.  Definitely do NOT want to bind to a variable here.
Based on our zoom meeting, I am changing these back to `_`